src/main/java/frc/robot/commands/TurnToGamepiece.java

```diff
@@ -0,0 +1,81 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.networktables.DoubleSubscriber;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.SwerveDrivetrain;
+import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
+
+public class TurnToGamepiece extends Command {
+    private SwerveDrivetrain drivetrain;
+    // private DoubleSupplier x_position;
+    // private DoubleSupplier y_position;
+    private DoubleSubscriber xSub;
+    private DoubleSubscriber ySub;
+    private double x;
+    private double y;
+    ChassisSpeeds driveStates;
+    private boolean finished = false;
+    private final ProfiledPIDController TurningPIDController;
+    private final ProfiledPIDController MovingPIDController;
+
+    private double cycle;
+
+    public TurnToGamepiece(SwerveDrivetrain drivetrain) {
+        this.drivetrain = drivetrain;
+        addRequirements(drivetrain);
+        NetworkTableInstance inst = NetworkTableInstance.getDefault();
+        NetworkTable table = inst.getTable("limelight");
```

Comment:
I think that all the Network Table readings should be done in a Vision subsystems that will read the values in its periodic method, and add getters for it to be used in the command.
This, for example, will allow us to know if there's a game piece in sight before we start the command.